app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, { FormEvent, useEffect, useRef, useState } from 'react';
+import { debounce } from 'lodash';
+import T from 'i18n-react';
+import { ActionType } from 'components/RecipeList/types';
+import {
+  ICreateRecipeProps,
+  IRecipeFormData,
+  IRecipeNameErrorData,
+} from 'components/RecipeManagement/types';
+import RecipeForm from 'components/RecipeManagement/RecipeForm';
+import MyDataPrepApi from 'api/dataprep';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import useFetch from 'services/react/customHooks/useFetch';
+
+const PREFIX = 'features.WranglerNewUI.RecipeForm.labels';
+const recipeNameRegEx = /^[a-z\d\s]+$/i;
```

Comment:
please add a proper comment to explain the regex expression with an example
also mention the use case w.r.t. out usage

Reply:
added comment which explains regex expression usecase with an example